File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        unordered_map<int, int> m;
        int pairs=0;
        for(int num: nums)
            m[num]++;
        for(auto item: m)
            pairs+=item.second*(item.second-1)/2;
        return pairs;
    }
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        freq = {}
        pairs = 0
        for x in nums:
            freq.setdefault(x, 0)
            freq[x] += 1
        for k, v in freq.items():
            pair = v * (v - 1) // 2
            pairs += pair
        return pairs
